matching/src/main/scala/org/kframework/backend/llvm/matching/Heuristics.scala

```diff
@@ -0,0 +1,61 @@
+package org.kframework.backend.llvm.matching
+
+import org.kframework.parser.kore.SymbolOrAlias
+import org.kframework.backend.llvm.matching.pattern._
+
+trait Heuristic {
+  def scoreAs[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, pat: Pattern[T]): Double
+  def scoreList[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean): Double
+  def scoreLiteral[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean): Double
+  def scoreMap[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, keys: Seq[Pattern[T]], frame: Option[Pattern[T]], canonicalize: Clause => MapP[Option[Occurrence]]): Double
+  def scoreOr[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, ps: Seq[Pattern[T]]): Double
+  def scoreSet[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, elements: Seq[Pattern[T]], frame: Option[Pattern[T]], canonicalize: Clause => SetP[Option[Occurrence]]): Double
+  def scoreSymbol[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, sym: SymbolOrAlias): Double
+  def scoreVariable[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean): Double
+  def scoreWildcard[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean): Double
+}
+
+class FHeuristic() extends Heuristic {
```

Comment:
let's call this DefaultHeuristic because it's definitely different than the F heuristic in the paper.